test.py

```diff
@@ -289,11 +289,11 @@ async def add_test(shortname, casename) -> None:
         if len(pending) == 0:
             return
         try:
-            await asyncio.gather(*pending)
+            await safe_gather(*pending)
         except asyncio.CancelledError:
             for task in pending:
                 task.cancel()
-            await asyncio.gather(*pending, return_exceptions=True)
```

Comment:
here return_exceptions=True in original asyncio.gather means waiting for ALL pending tasks finished with ignoring errors 

you can leave the original await asyncio.gather(*pending, return_exceptions=True)
it is also safe just not raise error :)

Reply:
I would prefer to have one method to handle gather rather than have the ability to use asyncio.gather() with return_exceptions=True. It will lead to confusion about when and what to use. It's better to rewrite the safe_gather() method to handle this situation. Maybe even rename it to something more adequate like gathert_wrapper() and always use this method when we need to use gather.